brave/brave-core
# Replace Rewards contributions P3A questions

Resolves https://github.com/brave/brave-browser/issues/29302
Resolves https://github.com/brave/brave-browser/issues/29301
Resolves https://github.com/brave/brave-browser/issues/29300



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0. Start with fresh profile, enable rewards. Ensure `Brave.Rewards.AutoContributionsState.3` and `Brave.Rewards.TipsSent` do not exist on local state page.
1. Connect to an external wallet. Ensure the auto contribute metric has the correct answer for the given auto contribute setting, and ensure the tips sent metric does not exist.
2. Disable auto contribute, ensure the relevant metric is updated accordingly.
3. Set two monthly tips and send two one time tips to verified creators. Wait 30 seconds. Ensure tip sent metric is set to 2.
4. Advance system time forward by 31 days and restart browser, ensure metric is set to 1, to note the 2 monthly tips only.

Ensure `Brave.Rewards.TipsState.2` and `Brave.Rewards.AutoContributionsState.2` no longer exist. Regression testing of `Brave.Rewards.AdsState.2` is recommended.
